app/main.py

```diff
@@ -1,25 +1,109 @@
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Any, Union
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Union[
+                "ChildrenSlideLimitationValidator",
+                "AdultSlideLimitationValidator"
+            ],
+            name: str
+    ) -> None:
+        self.protected_name = f"_{name}"
+
+    def __get__(
+            self,
+            instance: Union[
+                "ChildrenSlideLimitationValidator",
+                "AdultSlideLimitationValidator"
+            ],
+            owner: Any
+    ) -> bool:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: Union[
+                "ChildrenSlideLimitationValidator",
+                "AdultSlideLimitationValidator"
+            ],
+            value: int
+    ) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            value = False
+        else:
+            value = True
+
+        setattr(instance, self.protected_name, value)
```

Comment:
why do u need to rewrite a value ?)
u can just raise an exception when condition didn't pass